app/models/vote.rb

```diff
@@ -0,0 +1,29 @@
+  class Vote < ActiveRecord::Base
+  belongs_to :review
+  belongs_to :user
+
+  validates :user, presence: true
+  validates :review, presence: true
+  validates :value, numericality: { in: -1..1 }
+  validates_uniqueness_of :user_id, scope: :review_id
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.